2019-02-08  Erica Xie  <ericax@lnxsrv04.seas.ucla.edu>

	* lib/timespec.h (timespec_cmp):
	To avoid integer overflow with old times, 
	use comparison instead of subtraction.

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2019-02-08 01:20:39.285961000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec > b.tv_sec)
+    return 1; 
+  else if(a.tv_sec < b.tv_sec) 
+      return -1;
+  else 
+    return a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
